r2/r2/models/subreddit.py

```diff
@@ -540,6 +540,33 @@ def remove_moderator(self, user, **kwargs):
         hook = hooks.get_hook("subreddit.remove_moderator")
         hook.call(subreddit=self, user=user)
 
+        if user == self.top_moderator_account and len(self.moderators) > 1:
```

Comment:
This gets into some pretty heavy indenting, I think this is going to be a lot cleaner as a separate function. I'd just do something like this:

    # if the top mod is being removed, make sure the new one has full perms
    if user._id == self.moderators[0]:
        self._handle_mod_succession()

(There's no need to check that there's more than 1 moderator, that will be automatically handled by the list of potential successors being empty)

Then move the code inside here to a new function, which will save some indentation and make it a little easier overall.